### ISSUE TITLE
Separate lint needless_lifetimes for types with multiple lifetimes

### ISSUE THREAD

**Issue by antoyo** [rust-lang/rust-clippy#12495]
### What it does

I'd like to propose to make the lint `needless_lifetimes` only warn for references (and possibly types with a single lifetime) and add a new separate lint like `needless_lifetimes`, but that warns for types with multiple lifetimes (where at least one could be elided).

An alternative could to be not warn about `needless_lifetimes` when the lifetimes contain multiple characters (like `'stuff`).

### Advantage

For types with multiple lifetimes, it might help readability to keep the lifetimes in the code even if they could be elided.
So, by having a new separate lint, users who prefer to keep lifetimes on types with multiple lifetimes, they could disable this new lint and keep the ability to get warned about `needless_lifetimes` for simple cases.

### Drawbacks

None since the choice would be up to the user.

### Example

```rust
struct Test<'a, 'b> {
    a: &'a i32,
    b: &'b i32,
}

fn test<'a, 'b>(test: Test<'a, 'b>) -> &'a i32 {
    test.a
}
```

Could be written as:

```rust
fn test<'a>(test: Test<'a, '_>) -> &'a i32 {
    test.a
}
```

**Comment by m-rph** [comment]
@antoyo instead of providing a separate lint, I would like to introduce a configuration in this lint, allowing you to toggle this functionality as follows:

- both: `needless_lifetimes` will trigger for single LT and multiple LT
- single-lt: `needless_lifetimes` will trigger for single LT
- multiple-lt: `needless_lifetimes` will trigger only when multiple LTs exist but not for cases with a single LT that could be elided.

**Comment by antoyo** [comment]
I didn't know lints could be configured.
Are you talking about [this](https://doc.rust-lang.org/clippy/configuration.html)?
If so, there's this note:
> Note: The configuration file is unstable and may be deprecated in the future.

Not sure if this going to be a problem.

**Comment by m-rph** [comment]
We discussed this a bit on zulip and the consensus is that this lint and suggestion are good candidates for that.

**Comment by antoyo** [comment]
I'm good with this.
Thanks for working on this!
